File: 0383-ransom-note/0383-ransom-note.cs
public class Solution {
    public bool CanConstruct(string ransomNote, string magazine) {

      Dictionary< char, int> Hashmap = new Dictionary<char, int>();
      // char : key

      //first i am going to store num of each char in magazine, ie) char : count

      foreach(char c in magazine){
        if(!Hashmap.ContainsKey(c)){
          Hashmap[c] = 0;
        }
        Hashmap[c]++;
      }
      
      //now it has key and num of magazine's char

      foreach(char c in ransomNote){
        if(!Hashmap.ContainsKey(c) || Hashmap[c] == 0){
          return false;
        }
        Hashmap[c]--;
      }
      return true;
        
    }
}

File: 0383-ransom-note/0383-ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
      freq_map = {}

      for word in magazine:
        freq_map[word] = freq_map.get(word, 0) + 1

      for word in ransomNote:
        if word not in freq_map or freq_map[word] == 0:
          return False
        freq_map[word] -= 1
      return True